File: qc-core/src/main/java/com/eager/questioncloud/core/domain/coupon/infrastructure/CouponRepository.java
package com.eager.questioncloud.core.domain.coupon.infrastructure;

import com.eager.questioncloud.core.domain.coupon.model.Coupon;
import org.springframework.stereotype.Repository;

@Repository
public interface CouponRepository {
    Coupon findById(Long id);

    Coupon findByCode(String code);

    Coupon save(Coupon coupon);

    Boolean decreaseCount(Long couponId);
}

File: qc-payment/src/main/kotlin/com/eager/questioncloud/coupon/infrastructure/repository/CouponRepository.kt
package com.eager.questioncloud.coupon.infrastructure.repository

import com.eager.questioncloud.coupon.domain.Coupon
import org.springframework.stereotype.Repository

@Repository
interface CouponRepository {
    fun findById(id: Long): Coupon

    fun findByCode(code: String): Coupon

    fun save(coupon: Coupon): Coupon

    fun decreaseCount(couponId: Long): Boolean

    fun deleteAllInBatch()
}
